packages/adapters/src/adapters/Cornerstone3D/Segmentation/generateSegmentation.ts

```diff
@@ -1,22 +1,46 @@
-import { normalizers, derivations } from "dcmjs";
-import { fillSegmentation } from "../../Cornerstone/Segmentation_4X";
+import { normalizers, derivations } from 'dcmjs';
+import { Enums } from '@cornerstonejs/core';
+import { fillSegmentation } from '../../Cornerstone/Segmentation_4X';
 
-const { Normalizer } = normalizers;
+const { MetadataModules } = Enums;
+const { SEGImageNormalizer } = normalizers;
 const { Segmentation: SegmentationDerivation } = derivations;
 
+interface IOptions {
+  predecessorImageId?: string;
+}
+
+type Options = IOptions & {
+  [key: string]: number | boolean | unknown | string;
+};
+
 /**
  * generateSegmentation - Generates a DICOM Segmentation object given cornerstoneTools data.
  *
  * @param images - An array of the cornerstone image objects, which includes imageId and metadata
  * @param labelmaps - An array of the 3D Volumes that contain the segmentation data.
  */
-function generateSegmentation(images, labelmaps, metadata, options = {}) {
-    const segmentation = _createMultiframeSegmentationFromReferencedImages(
-        images,
-        metadata,
-        options
+function generateSegmentation(
+  images,
+  labelmaps,
+  metadata,
+  options: Options = {}
+) {
+  const segmentation = _createMultiframeSegmentationFromReferencedImages(
+    images,
+    metadata,
+    options
+  );
+  const segmentationResult = fillSegmentation(segmentation, labelmaps, options);
+  const predecessorImageId = options?.predecessorImageId;
```

Comment:
This provides a correct replacement reference when one segmentation replaces another.